runtime/vm/jnimisc.cpp

```diff
@@ -515,11 +515,12 @@ newStringUTF(JNIEnv *env, const char *bytes)
 			/* Slow case -- verify the encoding, fix if necessary.
 			 * String may double in size (invalid singles becoming doubles).
 			 */
-			data = (U_8*)jniArrayAllocateMemoryFromThread(currentThread, length * 2);
+			data = (U_8*)jniArrayAllocateMemoryFromThread(currentThread, length * 2 + 1);
 			if (NULL == data) {
 				gpCheckSetNativeOutOfMemoryError(currentThread, 0, 0);
 				goto done;
 			}
+			memset(data,0, length * 2 + 1);
```

Comment:
This memset is also not required as we're overwriting the data in the next step and all the apis are guaranteed to only look at `length` worth of data